src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -398,13 +398,21 @@ uct_rc_mlx5_iface_init_rx(uct_rc_iface_t *rc_iface,
     if (UCT_RC_MLX5_TM_ENABLED(iface)) {
         if (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX) {
             status = uct_rc_mlx5_devx_init_rx_tm(iface, rc_config, 0);
+            if (status != UCS_OK) {
+                return status;
+            }
         } else {
             status = uct_rc_mlx5_init_rx_tm(iface, rc_config, &srq_attr,
                                             UCT_RC_RNDV_HDR_LEN);
-        }
-
-        if (status != UCS_OK) {
-            return status;
+            if (status != UCS_OK) {
+                return status;
+            }
+
+            status = uct_ib_mlx5_srq_init(&iface->rx.srq, iface->rx.srq.verbs.srq,
```

Comment:
who sets `iface->rx.srq.type    = UCT_IB_MLX5_OBJ_TYPE_VERBS` for this case?
can we unite this with line 427..432, maybe by got or adding 'else' for lines 422..425?
or push it inside uct_rc_mlx5_init_rx_tm() to unite code with dc_mlx5?

Reply:
https://github.com/openucx/ucx/pull/3963/commits/b26052c1ab72ecd7401646ce2f3bda332e8957a1#diff-821e27f86a36190235dc4e13d03f23a2R789
each variant has its own flaws...